tools/clang/unittests/HLSLExec/ExecutionTest.cpp

```diff
@@ -11096,6 +11228,609 @@ TEST_F(ExecutionTest, PackUnpackTest) {
   }
 }
 
+// TODOLongVec : Need to change the members to vectors. But when I did that the
+// copy logic to read back from the shader buffer fails. Needs to fix that
+// before checking in PR.
+// SLongVectorBinaryOp is used in ShaderOpArithTable.xml. The shader program
+// uses the struct defintion to read from the input global buffer.
+template <typename T, std::size_t N,
+// By checking that T is an arithmetic we can keep the compiler error messages
+// from misusing the template much cleaner.
+typename = std::enable_if_t<std::is_arithmetic_v<T>>>
+struct SLongVectorBinaryOp {
+  T scalarInput;
+  std::array<T, N> vecInput1;
+  std::array<T, N> vecInput2;
+  std::array<T, N> vecOutput;
+};
+
+// vec1 == Expected vector
+// vec2 == Actual vector
+template <typename T, typename = std::enable_if_t<std::is_arithmetic_v<T>>>
+bool DoVectorsMatch(const std::vector<T>& vec1, const std::vector<T>& vec2, double tolerance) {
+    // Ensure both vectors have the same size
+    if (vec1.size() != vec2.size()) {
+      VERIFY_IS_TRUE(false, L"Vectors are of different sizes!");
+      return false;
+    }
+    
+    bool vectorsMatch = true;
+
+    // Stash mismatched indexes for easy failure logging later
+    std::vector<size_t> mismatchedIndexes;
+    for (size_t i = 0; i < vec1.size(); ++i) {
+      if (tolerance == 0 && vec1[i] != vec2[i]) {
+          vectorsMatch = false;
+          mismatchedIndexes.push_back(i); 
+      } else if constexpr (std::is_same_v<T, bool>) {
+        // Compiler was very picky and wanted an explicit case for bools to
+        // avoid a warning about comparing bools with >.
+        if (vec1[i] != vec2[i]) {
+            vectorsMatch = false;
+            mismatchedIndexes.push_back(i); 
+        }
+      } else {
+        T diff = vec1[i] > vec2[i] ? vec1[i] - vec2[i] : vec2[i] - vec1[i];
+        if (diff > tolerance) {
+          vectorsMatch = false;
+          mismatchedIndexes.push_back(i);
+        }
+      }
+    }
+
+    // Print the mismatched indexes and their corresponding elements
+    if (!mismatchedIndexes.empty()) {
+        for (size_t index : mismatchedIndexes) {
+          std::wstringstream wss(L"");
+          wss << L"Mismatch at Index: " << index;
+          wss << L" Expected Value:" << vec1[index];
+          wss << L" Actual Value:" << vec2[index];
+          WEX::Logging::Log::Error(wss.str().c_str());
+        }
+    }
+
+    return vectorsMatch;
```

Comment:
nit: Slightly different way of structuring. It combines Damyan's suggestion with applying the coding conventions for early exits (https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code).

```suggestion
    if (mismatchedIndexes.empty())
      return true;
      
    for (size_t index : mismatchedIndexes) {
      std::wstringstream wss(L"");
      wss << L"Mismatch at Index: " << index;
      wss << L" Expected Value:" << vec1[index];
      wss << L" Actual Value:" << vec2[index];
      WEX::Logging::Log::Error(wss.str().c_str());
    }

    return false;
```